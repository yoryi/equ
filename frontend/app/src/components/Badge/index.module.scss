:local(.container) {
    display: inline-block;
    padding: 5px 12px;
    border-radius: 6px;
    font-size: 12px;
    line-height: 18px;

    &:local(.green) {
        background-color: var(--green-10);
        color: var(--green);
    }

    &:local(.yellow) {
        background-color: var(--chromeYellow-20);
        color: var(--chromeYellow);
    }

    &:local(.red) {
        background-color: var(--light-red);
        color: var(--main-red);
    }

    &:local(.gray) {
        background-color: var(--light-grey);
        color: var(--grey);
    }
}
