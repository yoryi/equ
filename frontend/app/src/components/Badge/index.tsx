//Utils
import classNames from "classnames"
import * as React from "react"

//Styles
import Styles from "./index.module.scss"

interface BadgeProps {
  variant: "green" | "yellow" | "red" | "gray"
  children: React.ReactNode
  className?: string
}

const Badge: React.FC<BadgeProps> = ({ variant, children, className }) => (
  <div className={classNames(Styles.container, Styles[variant], className)}>
    {children}
  </div>
)

export default Badge
