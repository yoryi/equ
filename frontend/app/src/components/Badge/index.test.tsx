//Testing
import { render } from "@testing-library/react"
import faker from "faker"
import * as React from "react"

//Components
import Badge from "./"

describe(`Badge component`, () => {
  faker.seed(1000)

  describe(`it should render and match the snapshot`, () => {
    it(`green`, () => {
      const { container } = render(
        <Badge variant={`green`}>{faker.lorem.word()}</Badge>,
      )

      expect(container.firstChild).toMatchSnapshot()
    })

    it(`yellow`, () => {
      const { container } = render(
        <Badge variant={`yellow`}>{faker.lorem.word()}</Badge>,
      )

      expect(container.firstChild).toMatchSnapshot()
    })

    it(`red`, () => {
      const { container } = render(
        <Badge variant={`red`}>{faker.lorem.word()}</Badge>,
      )

      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
