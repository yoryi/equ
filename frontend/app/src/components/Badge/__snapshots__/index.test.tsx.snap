// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component it should render and match the snapshot green 1`] = `
<div
  class="container green"
>
  et
</div>
`;

exports[`Badge component it should render and match the snapshot red 1`] = `
<div
  class="container red"
>
  fugit
</div>
`;

exports[`Badge component it should render and match the snapshot yellow 1`] = `
<div
  class="container yellow"
>
  quia
</div>
`;
